core/templates/dev/head/components/entity-creation-services/topic-creation.service.ts

```diff
@@ -77,20 +78,19 @@ angular.module('oppia').factory('TopicCreationService', [
           AlertsService.clearWarnings();
 
           $rootScope.loadingMessage = 'Creating topic';
-          $http.post('/topic_editor_handler/create_new', {
-            name: topic.topicName,
-            abbreviated_name: topic.abbreviatedTopicName
-          }).then(function(response) {
-            $timeout(function() {
-              $window.location = UrlInterpolationService.interpolateUrl(
-                TOPIC_EDITOR_URL_TEMPLATE, {
-                  topic_id: response.data.topicId
-                }
-              );
-            }, 150);
-          }, function() {
-            $rootScope.loadingMessage = '';
-          });
+          TopicCreationBackendApiService.createTopic(
+            topic.topicName, topic.abbreviatedTopicName).then(
+            function(response) {
+              $timeout(function() {
+                $window.location = UrlInterpolationService.interpolateUrl(
+                  TOPIC_EDITOR_URL_TEMPLATE, {
+                    topic_id: response.topicId
+                  }
+                );
+              }, 150);
```

Comment:
Can you please check why we need a timeout here?

Reply:
I think it was added since there was a delay in between receiving the actual response and the topic being stored in the database? Not sure, but I remember adding a timeout somewhere due to this issue, not sure if this was the case here.
@abhayraghuwanshi Can you check if it works without the timeout (topic creation)? If so, you can remove it.

Reply:
It work fine after removing it so i removed it for now. Should i re-add it or not?